api/src/dtos/feature-flags/feature-flag-associate.dto.ts

```diff
@@ -0,0 +1,26 @@
+import { Expose, Type } from 'class-transformer';
+import { IsArray, IsDefined, IsString, IsUUID } from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
+import { IdDTO } from '../shared/id.dto';
+import { ValidationsGroupsEnum } from '../../enums/shared/validation-groups-enum';
+
+export class FeatureFlagAssociate {
+  @Expose()
+  @IsString({ groups: [ValidationsGroupsEnum.default] })
+  @IsUUID(4, { groups: [ValidationsGroupsEnum.default] })
+  @IsDefined({ groups: [ValidationsGroupsEnum.default] })
+  @ApiProperty()
+  id: string;
+
+  @Expose()
+  @Type(() => IdDTO)
+  @IsArray({ groups: [ValidationsGroupsEnum.default] })
+  @ApiProperty({ type: IdDTO, isArray: true })
+  associate: IdDTO[];
```

Comment:
I used IdDTO because I felt we will likely need the id AND name when using this endpoint instead of just sending the id. Ordinal is unnecessary but didn't see a better way to send both with the existing DTOs and this the pattern used elsewhere

Reply:
I wanted the validations on the id field which would be missing with a `string[]` being passed in. This is the patten I could find elsewhere

Reply:
Is the validation just that it is a UUID? Are you not able to do the validation on string because it's an array?

Reply:
Yes, just that it is a UUID. I couldn't find anywhere where we do uuid validation on an array. Can try to find a way to do nested validation